src/solvers/dgsem_structured/dg_2d.jl

```diff
@@ -418,4 +418,21 @@ function apply_jacobian!(du,
 end
 
 
+function total_volume(mesh::Union{StructuredMesh{2}, UnstructuredMesh2D, P4estMesh{2}}, dg, cache)
+  @unpack weights = dg.basis
+
+  total_volume = zero(real(mesh))
+
+  # Use quadrature to numerically integrate over entire domain
+  for element in eachelement(dg, cache)
```

Comment:
Does using this separate loop that again needs to calculate the volume Jacobian and then the total volume have a visible performance impact on the runtime of the analysis callback?

Reply:
especially in 3D?

Reply:
This is a benchmark of `analysis_callback(integrator)` of `examples/structured_3d/elixir_advection_basic.jl`, before:
```
BechmarkTools.Trial: 175 samples with 1 evaluations.
 Range (min … max):  17.879 ms … 49.269 ms  ┊ GC (min … max): 0.00% … 35.83%
 Time  (median):     27.454 ms              ┊ GC (median):    0.00%
 Time  (mean ± σ):   28.557 ms ±  4.101 ms  ┊ GC (mean ± σ):  0.35% ±  2.71%

                         █▆   ▁▁                               
  ▃▁▁▁▁▁▁▁▁▁▃▁▁▄▁▁▃▇▆▅▇▅████▆▆██▅▃▇█▅▆▆▄▄▁▅▄▅▅▁▄▅▆▅█▄▆▅▅▁▃▁▁▃ ▃
  17.9 ms         Histogram: frequency by time        37.4 ms <

 Memory estimate: 790.38 KiB, allocs estimate: 1457.
```
and after:
```
 BechmarkTools.Trial: 151 samples with 1 evaluations.
 Range (min … max):  20.883 ms … 57.388 ms  ┊ GC (min … max): 0.00% … 42.23%
 Time  (median):     31.011 ms              ┊ GC (median):    0.00%
 Time  (mean ± σ):   33.162 ms ±  6.293 ms  ┊ GC (mean ± σ):  0.48% ±  3.44%

                  █ ▅▃▃                                        
  ▃▁▁▃▃▁▁▃▄▁▃▅▅▃▇▆█████▆▇▅▅█▄▄▇▁▆▁▅▅▄▁▄▄▃▁█▅▆▃▄▁▃▁▅▆▁▄▄▃▄▃▁▁▄ ▃
  20.9 ms         Histogram: frequency by time        47.2 ms <

 Memory estimate: 790.84 KiB, allocs estimate: 1463.
```

Reply:
Hm, that's 16% slower - not super slow, but still measurable. Does it affect only the structured solver or all solvers?

Reply:
It should, in theory, affect all solvers with curvilinear meshes.

Reply:
OK, I am not sure about this at the moment, so let's leave it open for now.